### PR TITLE
feature: support nextval and currval

### LINKED ISSUE #675: [SEQUENCE] support CURRVAL,NEXTVAL for sequence in select statement
Scenario:
User can easily get a current/next sequence value over request a SQL below:

```sql
SELECT student.CURRVAL, student.NEXTVAL
```

Then, a row with current/next sequence value of `student` will be returned.
If the sequence `student` doesn't exist, an error will be returned.

Notice:
1. only match those SQLs contain `<sequence_name>.[CURRVAL|NEXTVAL]`, and no `FROM` followed.
2. use internal sequence API as implementation.
3. should handle alias, for example, `select student.NEXTVAL as xxxx`.
4. ...

### PR DESCRIPTION
**Pull request by Lvnszn**
**What this PR does**:

- support currval and next val.
- support alias in currval and nextval

**Which issue(s) this PR fixes**:

Fixes #675

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/runtime/plan/dml/local_sequence.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dml
+
+import (
+	"context"
+	rcontext "github.com/arana-db/arana/pkg/runtime/context"
```
Comment (dongzl) [inline pkg/runtime/plan/dml/local_sequence.go:22]: please format import, use this tool:
```
go get -u github.com/dubbogo/tools/cmd/imports-formatter

imports-formatter
```

File: test/integration_test.go [hunk 2]
```diff
@@ -1376,3 +1379,60 @@ func (s *IntegrationSuite) TestExplain() {
 		})
 	}
 }
```
Comment (dongzl) [inline test/integration_test.go:1381]: same

**Review by dongzl — CHANGES_REQUESTED** [review]

**Review by maronghe — APPROVED** [review]
LGTM

File: pkg/runtime/optimize/dml/select.go [hunk 3]
```diff
@@ -94,9 +98,26 @@ func optimizeSelect(ctx context.Context, o *optimize.Optimizer) (proto.Plan, err
 				}
 				valueList = append(valueList, calculateRes)
 				columnList = append(columnList, stmt.Select[i].DisplayName())
-
+			case *ast.SelectElementColumn:
+				if len(selectItem.Name) == 2 && (strings.ToLower(selectItem.Name[1]) == "currval" || strings.ToLower(selectItem.Name[1]) == "nextval") {
```
Comment (jjeffcaii) [inline pkg/runtime/optimize/dml/select.go:102]: Use `strings.EqualFold` instead.

**Review by jjeffcaii — APPROVED** [review]

File: pkg/runtime/optimize/dml/select.go [hunk 4]
```diff
@@ -94,9 +98,27 @@ func optimizeSelect(ctx context.Context, o *optimize.Optimizer) (proto.Plan, err
 				}
 				valueList = append(valueList, calculateRes)
 				columnList = append(columnList, stmt.Select[i].DisplayName())
-
+			case *ast.SelectElementColumn:
```
Comment (Mulavar) [inline pkg/runtime/optimize/dml/select.go:101]: Should we detect if there's from clause followed?

File: test/integration_test.go [hunk 5]
```diff
@@ -1376,3 +1379,60 @@ func (s *IntegrationSuite) TestExplain() {
...
+		{
+			"select student.currVal",
+			1,
+		},
+		{
+			"select student.nextVal",
+			2,
+		},
+		{
+			"select notexist.currVal",
```
Comment (Mulavar) [inline test/integration_test.go:1420]: ditto. if the user executes sql like 'select student.currVal from xxx',  do we expect some error?

File: pkg/runtime/optimize/dml/select.go (hunk 4 above)
Comment (Lvnszn) [reply pkg/runtime/optimize/dml/select.go:101]: As stated in the issue, no FROM followed.

File: test/integration_test.go (hunk 5 above)
Comment (Lvnszn) [reply test/integration_test.go:1420]: I think there is no need to report an error, because there may be real query scenarios to query this field

**Review by dongzl — APPROVED** [review]

File: pkg/runtime/optimize/dml/select.go (hunk 4 above)
Comment (Mulavar) [reply pkg/runtime/optimize/dml/select.go:101]: Yeah, I know it, but if I write this sql with from followed which is a wrong SQL, should I report an error?

**Pull request merged**